src/main/java/ch/njol/skript/expressions/ExprExplosionPrimeEventYield.java

```diff
@@ -0,0 +1,119 @@
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.expressions;
+
+import org.bukkit.event.Event;
+import org.bukkit.event.entity.ExplosionPrimeEvent;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.ScriptLoader;
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Events;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.log.ErrorQuality;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+
+@Name("Explosion Prime Event Explosion Yield")
+@Description("The yield/radius of the explosion in an explosion prime event.")
+@Events("explosion prime")
+@Examples("set the explosion radius to 10")
+@Since("INSERT VERSION")
+public class ExprExplosionPrimeEventYield extends SimpleExpression<Number> {
+
+	static {
+		Skript.registerExpression(ExprExplosionPrimeEventYield.class, Number.class, ExpressionType.PROPERTY, "[the] explosion (radius|size|yield)");
+	}
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (!ScriptLoader.isCurrentEvent(ExplosionPrimeEvent.class)) {
+			Skript.error("The explosion radius is only usable in explosion prime events", ErrorQuality.SEMANTIC_ERROR);
+			return false;
+		}
+		return true;
+	}
+
+	@Override
+	@Nullable
+	protected Number[] get(Event e) {
+		return CollectionUtils.array(((ExplosionPrimeEvent) e).getRadius());
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(final ChangeMode mode) {
+		if (mode == ChangeMode.REMOVE_ALL || mode == ChangeMode.RESET)
+			return null;
+		return CollectionUtils.array(Number.class);
+	}
+
+	@Override
+	public void change(final Event event, final @Nullable Object[] delta, final ChangeMode mode) {
+		float f = delta == null ? 0 : ((Number) delta[0]).floatValue();
+		ExplosionPrimeEvent e = (ExplosionPrimeEvent) event;
+		switch (mode) {
+			case SET:
+				e.setRadius(f);
+				break;
+			case ADD:
+				float add = e.getRadius() + f;
+				if (add < 0) add = 0;
+				e.setRadius(add);
+				break;
+			case REMOVE:
+				float subtract = e.getRadius() - f;
+				if (subtract < 0) subtract = 0;
+				e.setRadius(subtract);
+				break;
+			case DELETE:
+				e.setRadius(0);
+				break;
+			case REMOVE_ALL:
+				assert false;
+			case RESET:
+				assert false;
```

Comment:
Correct me if I'm wrong, but since this is a void method, I don't think this is needed here.

Reply:
it's syntactically valid in a void method - asset doesn't return it just gives some ease to debugging when you use `-ea`. but yeah not needed here, any caller should be checking the `allowChange` method prior to calling `change`

Reply:
I added in the other cases to remove the warning, but I can remove them